packages/jest-cli/src/__tests__/watch.test.js

```diff
@@ -175,26 +186,46 @@ describe('Watch mode flows', () => {
     expect(pipeMockCalls.slice(determiningTestsToRun + 1)).toMatchSnapshot();
   });
 
-  it('triggers enter on a WatchPlugin when its key is pressed', () => {
-    const plugin = require(watchPluginPath);
+  it('triggers enter on a WatchPlugin when its key is pressed', async () => {
+    const showPrompt = jest.fn(() => Promise.resolve());
+    const pluginPath = `${__dirname}/__fixtures__/plugin_path`;
+    jest.doMock(
+      pluginPath,
+      () =>
+        class WatchPlugin1 {
+          constructor() {
+            this.showPrompt = showPrompt;
+          }
+          getUsageRow() {
+            return {
+              key: 's'.codePointAt(0),
+              prompt: 'do nothing',
+            };
+          }
+        },
+      {virtual: true},
+    );
 
     watch(
       Object.assign({}, globalConfig, {
         rootDir: __dirname,
-        watchPlugins: [watchPluginPath],
+        watchPlugins: [pluginPath],
       }),
       contexts,
       pipe,
       hasteMapInstances,
       stdin,
     );
 
-    stdin.emit(plugin.key.toString(16));
+    stdin.emit(Number('s'.charCodeAt(0)).toString(16));
+    // showPrompt();
+    await nextTick();
 
-    expect(plugin.enter).toHaveBeenCalled();
+    expect(showPrompt).toHaveBeenCalled();
   });
 
-  it('prevents Jest from handling keys when active and returns control when end is called', () => {
+  // TODO: Fix or remove before merging
+  xit('prevents Jest from handling keys when active and returns control when end is called', () => {
```

Comment:
seems like something we'd want, isn't it?

Reply:
Correct, this is one of the pending items that I have at the top.